src/Raven.Studio/typescript/components/shell/chatbot/store/chatbotSlice.ts

```diff
@@ -186,8 +266,14 @@ export const chatbotSelectors = {
     isPinned: (state: RootState) => state.chatbot.isPinned,
     chatbotTab: (state: RootState) => state.chatbot.chatbotTab,
     chatbotResourcesTab: (state: RootState) => state.chatbot.chatbotResourcesTab,
-    messages: (state: RootState) => state.chatbot.messages,
+    messageIds: (state: RootState) => chatbotMessagesSelectors.selectIds(state.chatbot.messages),
+    messageById: (state: RootState, id: string) => chatbotMessagesSelectors.selectById(state.chatbot.messages, id),
+    messagesCount: (state: RootState) => chatbotMessagesSelectors.selectTotal(state.chatbot.messages),
+    isLastMessage: (state: RootState, id: string) => {
+        const messageIds = chatbotMessagesSelectors.selectIds(state.chatbot.messages);
+        return messageIds[messageIds.length - 1] === id;
```

Comment:
there is nothing wrong with that, but consider using `.at(-1)`, i think its clearer
```suggestion
        return messageIds.at(-1) === id;
```